### PR TITLE
Azure support

### LINKED ISSUE #455: Add support for azure

### PR DESCRIPTION
**Pull request by kron4eg**
Fixes #455 

```release-note
Azure support
```

### CODE REVIEW

File: examples/terraform/azure/variables.tf [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+variable "ssh_agent_socket" {
+  description = "SSH Agent socket, default to grab from $SSH_AUTH_SOCK"
+  default     = "env:SSH_AUTH_SOCK"
+}
+
+# Provider specific settings
+
+variable "location" {
+  description = ""
```
Comment (xmudrii) [inline examples/terraform/azure/variables.tf]: Are those descriptions missing intentionally? Should we add descriptions?

Comment (kron4eg) [reply examples/terraform/azure/variables.tf]: Apparently just missed it 🤷‍♂️

**Comment by kron4eg** [comment]
@xmudrii I've addressed (I think all) those comments, PTAL again.

File: pkg/apis/kubeone/validation/validation.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ func ValidateCloudProviderSpec(p kubeone.CloudProviderSpec, fldPath *field.Path)
 
 	switch p.Name {
 	case kubeone.CloudProviderNameAWS:
+	case kubeone.CloudProviderNameAzure:
```
Comment (xmudrii) [inline pkg/apis/kubeone/validation/validation.go]: Is `cloudConfig` required for Azure? If yes, should we enforce it here?

Comment (kron4eg) [reply pkg/apis/kubeone/validation/validation.go]: yeah... actually it's required for vSphere too and I don't remember that I've done it 😅

Comment (kron4eg) [reply pkg/apis/kubeone/validation/validation.go]: done

**Comment by kron4eg** [comment]
/override pull-kubeone-e2e-digitalocean-conformance-1.13.5

DO API strikes again: 503 Service is currently unavailable.

File: docs/quickstart-azure.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+values with real values.
+
+```yaml
+apiVersion: kubeone.io/v1alpha1
+kind: KubeOneCluster
+
+versions:
+  kubernetes: '1.14.2'
+cloudProvider:
+  name: 'vsphere'
```
Comment (craiglpeters) [inline docs/quickstart-azure.md]: This should say 'azure'

File: docs/quickstart-azure.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+Before deleting a cluster you should clean up all MachineDeployments, so all
+worker nodes are deleted. You can do it with the `kubeone reset` command:
+
+```bash
+kubeone reset config.yaml --tfjson tf.json
+```
+
+This command will wait for all worker nodes to be gone. Once it's done you can
+proceed and destroy the vSphere infrastructure using Terraform:
```
Comment (craiglpeters) [inline docs/quickstart-azure.md]: Another copy/paste error. Replace 'vSphere' with 'Azure'

**Pull request merged**